src/odemis/util/img.py

```diff
@@ -811,3 +813,107 @@ def getBoundingBox(content):
 
     return rect
 
+
+def integrateImages(imgs, n, integration_step, best_dtype=None):
```

Comment:
Honestly, this function usage is terrible. I'm sorry suggesting you to do something like that. You pass a lot of parameters, some of them are used only when other parameters are as special value. It also constantly returns a dtype, but you have to store it anyway. I think that we can do much better for the use cases by creating a class. The main difference is that the `best_dtype` and `integration_step` don't need to be stored apart. It'd look like this.
``` python
class ImageIntegrator(...):
    def __init__(steps):
         self.steps = steps  # can be changed by the caller, on the fly
         self._step = 0
         self._img = None
         self._best_dtype = None

    def append(img):
        """
        Integrate the image with the previous images and return the complete image.
        It will automatically reset the integrated image after reaching the number of integration count.
        """
       self._step += 1
        # if no image yet, do fist image code
        # else, add to _img
        # if reached the number of steps, do final integration, and reset all the values

      return img
```

This way, it should be possible to only store a "ImageIntegrator" in the two callers above, and nothing else really.